runtime/executor/src/main/java/edu/snu/nemo/runtime/executor/data/partitioner/HashPartitioner.java

```diff
@@ -40,6 +40,11 @@ public HashPartitioner(final int dstParallelism,
 
   @Override
   public Integer partition(final Object element) {
-    return Math.abs(keyExtractor.extractKey(element).hashCode() % dstParallelism);
+    final Object key = keyExtractor.extractKey(element);
+    if (key == null) {
```

Comment:
What's causing the key to be `null`?
I'd think changing `BeamKeyExtractor` to `return 0` would be better than changing `HashPartitioner`, as this behavior seems to be specific to Beam. (and does not apply to Spark)

Reply:
Beam WriteFiles produces KV.of(null, something), during the sink. I will change BeamKeyExtractor to return 0 for null key pairs.